<a href="https://colab.research.google.com/github/ksuaray/LAEP_S24/blob/Homelessness/Homelessness_Assignment_1_NB.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**National Estimates of Homelessness**

#**Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that enable us to be able to

In [2]:
import pandas as pd
import numpy as np
import plotly.express as px
from IPython.display import Image


#**Context**

The Continuum of Care (CoC) Program is designed to promote communitywide commitment to the goal of ending homelessness; provide funding for efforts by nonprofit providers, and State and local governments to quickly rehouse homeless individuals and families while minimizing the trauma and dislocation caused to homeless individuals, families, and communities by homelessness; promote access to and effect utilization of mainstream programs by homeless individuals and families; and optimize self-sufficiency among individuals and families experiencing homelessness. For more information on the Program, please visit https://www.hudexchange.info/programs/coc/

The U.S. Department of Housing and Urban Development (HUD) provides Point-in-Time (PIT) count reports of sheltered and unsheltered persons experiencing homelessness, by household type and subpopulation. This data is available at the national and state level, and for each CoC. HUD also provides Housing Inventory Count (HIC) reports, which provide a snapshot of a CoC’s inventory of beds and units available on the night designated for the count by program type, and include beds dedicated to serve persons who are homeless as well as persons in Permanent Supportive Housing.
This raw data set contains PIT estimates of homelessness, and the corresponding accompanying HIC data from 2021.

Attribution: Adapted from U.S. Department of Housing and Urban Development

#**About the Dataset**

This dataset contains 100 rows corresponding to a random sample of localities (typically counties or similar large regions) that received Continuum of Care funding from HUD. A total of 21 variables are provided as listed on the next page.

**Variables**

| Variable Name(s) | Description |
| --- | --- |
| "CoC Number<br>CoC Name" | CoC locality identifier |
| CoC Category | Community setting designation: Rural, Urban, etc |
| Type of Count | Description of whether fully or partially sheltered individuals (or both) are included |
| Overall Homeless, 2021 | Total homeless count in all facilities indicated by the “Type of Count” variable |
| HMIS Participation Rate for Year-Round Beds (ES,TH,SH) | Percentage of facilities in the locality that participate in the Homeless Management Information System (HMIS) |
| "Total Year-Round Beds (ES, TH, SH)<br>Total Year-Round Beds (ES)<br>Total Year-Round Beds (TH)<br>Total Year-Round Beds (SH)" | "(all types combined)<br>ES = Emergency Shelter<br>TH = Transitional Housing<br>SH = Safe Haven" |
| "Total Units for Households with Children (ES, TH, SH)<br>Total Beds for Households with Children (ES, TH, SH)" | Number of units and beds designated for households with children |
| Sheltered ES Homeless 2021 | Estimate of the number of individuals sheltered in an Emergency Facility at the time of the study |
| Sheltered ES Homeless - Age 18 to 24, 2021 | Estimate of the number of individuals sheltered in an Emergency Facility by age |
| "Sheltered ES Homeless - Female, 2021<br>Sheltered ES Homeless - Male, 2021<br>Sheltered ES Homeless - Trans+, 2021" | Estimate of the number of individuals sheltered in an Emergency Facility by gender |
| "Sheltered ES Homeless - Hispanic/Latino, 2021<br>Sheltered ES Homeless - White, 2021<br>Sheltered ES Homeless - Black or African American, 2021<br>Sheltered ES Homeless - Asian or Pacific Islander<br>Sheltered ES Homeless - American Indian or Alaska Native, 2021<br>Sheltered ES Homeless - Multiple Races, 2021" | Estimate of the number of individuals sheltered in an Emergency Facility by race/ethnicity |



We can view a snippet of the data by first importing it directly from the [url below](https://raw.githubusercontent.com/ksuaray/LAEP_S24/Homelessness/Homelessness_Data.csv).

**Data**

In [3]:
file_url = "https://raw.githubusercontent.com/ksuaray/LAEP_S24/Homelessness/Homelessness_Data.csv"
df = pd.read_csv(file_url)


Next, we can display the data by typing the name of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [4]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)
df

CoC Number                                           CoC Name  \
0      LA-505                     Monroe/Northeast Louisiana CoC   
1      MN-502                  Rochester/Southeast Minnesota CoC   
2      PA-508                     Scranton/Lackawanna County CoC   
3      OK-503                      Oklahoma Balance of State CoC   
4      OH-504                     Youngstown/Mahoning County CoC   
..        ...                                                ...   
95     KS-507                        Kansas Balance of State CoC   
96     CA-507                                   Marin County CoC   
97     TN-512  Morristown/Blount, Sevier, Campbell, Cocke Cou...   
98     MA-515                                     Fall River CoC   
99     CA-526  Amador, Calaveras, Mariposa, Tuolumne Counties...   

               CoC Category                   Type of Count  \
0         Largely Rural CoC            Sheltered-Only Count   
1         Largely Rural CoC            Sheltered-Only Count   
2      Largely Suburban CoC  Sheltered and full unsheltered   
3         Largely Rural CoC            Sheltered-Only Count   
4   Other Largely Urban CoC            Sheltered-Only Count   
..                      ...                             ...   
95        Largely Rural CoC            Sheltered-Only Count   
96     Largely Suburban CoC            Sheltered-Only Count   
97        Largely Rural CoC  Sheltered and full unsheltered   
98     Largely Suburban CoC  Sheltered and full unsheltered   
99        Largely Rural CoC            Sheltered-Only Count   

    Overall Homeless, 2021  Total Year-Round Beds (ES, TH, SH)  \
0                       78                                 133   
1                      419                                 599   
2                      165                                 155   
3                      125                                 149   
4                       62                                 198   
..                     ...                                 ...   
95                     783                                1170   
96                     225                                 293   
97                     472                                 133   
98                     324                                 236   
99                     151                                 214   

    HMIS Participation Rate for Year-Round Beds (ES, TH, SH)  \
0                                               0.233          
1                                               0.604          
2                                               0.839          
3                                               0.295          
4                                               0.192          
..                                                ...          
95                                              0.162          
96                                              0.727          
97                                              0.316          
98                                              0.877          
99                                              0.598          

    Total Year-Round Beds (ES)  Total Year-Round Beds (TH)  \
0                           69                          64   
1                          378                         221   
2                           64                          80   
3                          135                          14   
4                          187                          11   
..                         ...                         ...   
95                         703                         452   
96                         163                         130   
97                         117                          16   
98                         212                          24   
99                         156                          58   

    Total Year-Round Beds (SH)  \
0                            0   
1                            0   
2                   

#**ASSIGNMENT 1 - Descriptive Statistics: Graphical and Numerical Summary**

**INSTRUCTIONS**

Use Python to analyze the data set and complete each of the following. As appropriate, copy the SPSS output and paste it in the correct part below. For problems that require a written response, type the answer below.

##**QUESTION 1**

Determine whether the three variables below are qualitative or quantitative. If they are quantitative, specify whether they are continuous or discrete.

| Variable         | Qual or Quant | Dis, Con, or Neither |
|------------------|---------------|----------------------|
| **CoC Category**              | Qual or Quant  | Dis, Con, or Neither              |
| **HMIS Participation Rate for Year-Round Beds (ES,TH,SH)**           | Qual or Quant   | Dis, Con, or Neither              |
| **Sheltered ES Homeless - Age 18 to 24, 2021**           | Qual or Quant  | Dis, Con, or Neither           |


**CoC Category: Qualitative**

**HMIS Participation Rate For Year-Round Beds (ES,TH,SH): QUANTITATIVE CONTINUOUS**

**Sheltered ES Homeless - Age 18 to 24, 2021: QUANTITATIVE DISCRETE**

##**QUESTION 2**

Construct a frequency table, relative frequency table, and bar chart to describe the distribution of the following variable based on your last name (pick the variable based on your last name). State any fact that jumps out to you.

| Last Name Variable | Description |
| --- | --- |
| A-M | CoC Category |
| N-Z | Type Of Count |


In [14]:
#Frequency table - CoC Category
freq_table_coc_category = pd.value_counts(df['CoC Category'])
freq_table_coc_category

Largely Suburban CoC       43
Largely Rural CoC          33
Other Largely Urban CoC    13
Major City CoC             11
Name: CoC Category, dtype: int64

In [15]:
#Relative frequency table - CoC Category
freq_table_coc_category/len(df)

Largely Suburban CoC       0.43
Largely Rural CoC          0.33
Other Largely Urban CoC    0.13
Major City CoC             0.11
Name: CoC Category, dtype: float64

In [17]:
#Bar Chart - CoC Category
dfrf = pd.DataFrame(freq_table_coc_category)
fig = px.bar(x=dfrf.index, y=dfrf['CoC Category'], barmode='group',
             title='Frequency Distribution Bar Chart - CoC Category')
fig.show()

In [19]:
#Frequency table - Type Of Count
freq_table_type_of_count = pd.value_counts(df['Type of Count'])
freq_table_type_of_count

Sheltered-Only Count                 43
Sheltered and full unsheltered       33
Sheltered and partial unsheltered    24
Name: Type of Count, dtype: int64

In [20]:
#Relative frequency table - Type Of Count
freq_table_type_of_count/len(df)

Sheltered-Only Count                 0.43
Sheltered and full unsheltered       0.33
Sheltered and partial unsheltered    0.24
Name: Type of Count, dtype: float64

In [22]:
#Bar Chart - Type Of Count
dfrf = pd.DataFrame(freq_table_type_of_count)
fig = px.bar(x=dfrf.index, y=dfrf['Type of Count'], barmode='group',
             title='Frequency Distribution Bar Chart - Type Of Count')
fig.show()

## **QUESTIONS 3-6**

For questions 3-6: Find your variable based on your last name and use that variable when answering questions #3 to #6.  

| Last Name | Variable |
| --- | --- |
| A-F | Sheltered ES Homeless - Hispanic/Latino, 2021 |
| G-M | Sheltered ES Homeless - White, 2021 |
| N-S | Sheltered ES Homeless - Black or African American, 2021 |
| T-Z | Sheltered ES Homeless - Asian or Pacific Islander |



###**QUESTION 3**

Construct a histogram for your variable. Use Number of Intervals = 12.

In [32]:
# Histogram - Sheltered ES Homeless - Hispanic/Latino, 2021
fig = px.histogram(x=df['Sheltered ES Homeless - Hispanic/Latino, 2021'], labels = {'x':'Sheltered ES Homeless - Hispanic/Latino, 2021', 'y':'Frequency'},nbins = 12)
fig.show()

In [33]:
# Histogram - Sheltered ES Homeless - White, 2021
fig = px.histogram(x=df['Sheltered ES Homeless - White, 2021'],labels = {'x':'Sheltered ES Homeless - White, 2021', 'y':'Frequency'},nbins = 12)
fig.show()

In [34]:
# Histogram - Sheltered ES Homeless - Black or African American, 2021
fig = px.histogram(x=df['Sheltered ES Homeless - Black or African American, 2021'],labels = {'x':'Sheltered ES Homeless - Black or African American, 2021', 'y':'Frequency'},nbins = 12)
fig.show()

In [43]:
# Histogram - Sheltered ES Homeless - Asian or Pacific Islander
fig = px.histogram(x=df['Sheltered ES Homeless - Asian or Pacific Islander'],labels = {'x':'Sheltered ES Homeless - Asian or Pacific Islander', 'y':'Frequency'},nbins = 12)
fig.show()

###**QUESTION 4**

Construct a boxplot for your variable.  

In [39]:
# Box Plot - Sheltered ES Homeless - Hispanic/Latino, 2021
px.box(y=df['Sheltered ES Homeless - Hispanic/Latino, 2021'], title='Sheltered ES Homeless - Hispanic/Latino, 2021')

In [40]:
# Box Plot - Sheltered ES Homeless - White, 2021
px.box(y=df['Sheltered ES Homeless - White, 2021'], title='Sheltered ES Homeless - White, 2021')

In [41]:
# Box Plot - Sheltered ES Homeless - Black or African American, 2021
px.box(y=df['Sheltered ES Homeless - Black or African American, 2021'], title='Sheltered ES Homeless - Black or African American, 2021')

In [42]:
# Box Plot - Sheltered ES Homeless - Asian or Pacific Islander
px.box(y=df['Sheltered ES Homeless - Asian or Pacific Islander'], title='Sheltered ES Homeless - Asian or Pacific Islander')

###**QUESTION 5**

Find the following summary statistics for your variable from #3: minimum, maximum, mean, median, standard deviation, Q1, and Q3.

In [44]:
# Summary - Sheltered ES Homeless - Hispanic/Latino, 2021
df[['Sheltered ES Homeless - Hispanic/Latino, 2021']].describe(include='all')

Sheltered ES Homeless - Hispanic/Latino, 2021
count                                      100.00000
mean                                       135.34000
std                                        570.09367
min                                          0.00000
25%                                          6.50000
50%                                         21.50000
75%                                         57.25000
max                                       5511.00000

In [45]:
# Summary - Sheltered ES Homeless - White, 2021
df[['Sheltered ES Homeless - White, 2021']].describe(include='all')

Sheltered ES Homeless - White, 2021
count                           100.000000
mean                            313.270000
std                             748.266115
min                               0.000000
25%                              48.750000
50%                             104.500000
75%                             227.500000
max                            6628.000000

In [46]:
# Summary - Sheltered ES Homeless - Black or African American, 2021
df[['Sheltered ES Homeless - Black or African American, 2021']].describe(include='all')

Sheltered ES Homeless - Black or African American, 2021
count                                         100.000000      
mean                                          226.640000      
std                                           715.114578      
min                                             0.000000      
25%                                            22.500000      
50%                                            61.500000      
75%                                           133.250000      
max                                          6654.000000

In [47]:
# Summary - Sheltered ES Homeless - Asian or Pacific Islander, 2021
df[['Sheltered ES Homeless - Asian or Pacific Islander']].describe(include='all')

Sheltered ES Homeless - Asian or Pacific Islander
count                                         100.000000
mean                                           14.370000
std                                            40.297293
min                                             0.000000
25%                                             1.000000
50%                                             2.500000
75%                                             7.250000
max                                           274.000000

###**QUESTION 6**

Use information from questions #3, #4, and #5 to describe your variable in terms of shape, center, spread, and outliers. Interpret your findings.

NOTE: OUTLIERS ARE HARD TO COUNT, GRADE WITH A +/-4 TOLERANCE

THE DISTRIBUTION OF **Sheltered ES Homeless - Hispanic/Latino, 2021** IS SKEWED RIGHT. THE MEDIAN IS 21.50 PEOPLE AND THE IQR IS 60.25. THERE ARE 12 OUTLIERS IN THE RIGHT TAIL.


THE DISTRIBUTION OF **Sheltered ES Homeless - White, 2021** IS SKEWED RIGHT. THE MEDIAN IS 104.50 PEOPLE AND THE IQR IS 200.25. THERE ARE 11 OUTLIERS IN THE RIGHT TAIL.


THE DISTRIBUTION OF **Sheltered ES Homeless - Black or African American, 2021** IS SKEWED RIGHT. THE MEDIAN IS 61.50 PEOPLE AND THE IQR IS 126.25. THERE ARE 11 OUTLIERS IN THE RIGHT TAIL.


THE DISTRIBUTION OF **Sheltered ES Homeless - Asian or Pacific Islander** IS SKEWED RIGHT. THE MEDIAN IS 2.50 AND THE IQR IS 6.75. THERE ARE 14 OUTLIERS IN THE RIGHT TAIL.

##**QUESTION 7**

Calculate and state the mean Sheltered ES Homeless - White, 2021 for Major city CoC’s. Do the same for Largely Rural CoC’s. Compare the results.

---

**Report**

Sheltered ES Homeless - White, 2021

| CoC Category | Mean | N |
| --- | --- | --- |
| Largely Rural CoC | 175.55 | 33 |
| Largely Suburban CoC | 184.23 | 43 |
| Major City CoC | 1385.45 | 11 |
| Other Largely Urban CoC | 182.46 | 13 |
| Total | 313.27 | 100 |



##**QUESTION 8**

Generate a paragraph of at least 100 words to address one of the following questions:


### **QUESTION 8a**

Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

### **QUESTION 8b**

Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?

### **QUESTION 8c**

Discuss how analyzing your chosen data set using statistical methods could help you be aware of social issues, contribute to society, and advocate for marginalized communities.
